test/Microsoft.TemplateEngine.TemplateLocalizer.EndToEndTests/ExportCommandTests.cs

```diff
@@ -0,0 +1,207 @@
+    ""$schema"": ""http://json.schemastore.org/template"",
+    ""author"": ""Microsoft"",
+    ""classifications"": [""Common"", ""Library""],
+    ""name"": ""Class library"",
+    ""generatorVersions"": ""[1.0.0.0-*)"",
+    ""description"": ""dEscRiPtiON: ,./|\\<>{}!@#$%^&*()_+-=? 12 äÄßöÖüÜçÇğĞıIİşŞ"",
+    ""symbols"": {
+        ""TargetFrameworkOverride"": {
+          ""type"": ""parameter"",
+          ""displayName"": ""tfm display name"",
+          ""description"": ""tfm description"",
+          ""replaces"": ""TargetFrameworkOverride"",
+          ""datatype"": ""string"",
+          ""defaultValue"": """"
+        },
+        ""Framework"": {
+          ""type"": ""parameter"",
+          ""displayName"": ""framework display name"",
+          ""description"": ""framework description"",
+          ""datatype"": ""choice"",
+          ""choices"": [
+            {
+              ""choice"": ""net5.0"",
+              ""displayName"": ""net5.0 display name"",
+              ""description"": ""Target net5.0""
+            },
+            {
+              ""choice"": ""netstandard2.1"",
+              ""description"": ""Target netstandard2.1""
+            },
+            {
+              ""choice"": ""netstandard2.0"",
+              ""displayName"": ""netstandard2.0 display name"",
+              ""description"": ""Target netstandard2.0""
+            }
+          ],
+          ""replaces"": ""net5.0"",
+          ""defaultValue"": ""net5.0""
+        },
+    },
+    ""postActions"": [
+        {
+          ""condition"": ""(!skipRestore)"",
+          ""description"": ""Restore NuGet packages required by this project."",
+          ""manualInstructions"": [
+            {
+                ""text"": ""Run 'dotnet restore'""
+            }
+          ],
+          ""actionId"": ""210D431B-A78B-4D2F-B762-4ED3E3EA9025"",
+          ""continueOnError"": true
+        },
+        {
+        ""condition"": ""(HostIdentifier != \""dotnetcli\"" && HostIdentifier != \""dotnetcli-preview\"")"",
+          ""description"": ""Opens Class1.cs in the editor"",
+          ""manualInstructions"": [ ],
+          ""actionId"": ""84C0DA21-51C8-4541-9940-6CA19AF04EE6"",
+          ""args"": {
+            ""files"": ""1""
+          },
+          ""continueOnError"": true
+        }
+      ]
+}
+";
+
+        private string _workingDirectory;
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            _workingDirectory = Path.Combine(Path.GetTempPath(), "Microsoft.TemplateEngine.TemplateLocalizer.EndToEndTests", Path.GetRandomFileName());
+            Directory.CreateDirectory(_workingDirectory);
+        }
+
+        [TestCleanup]
+        public void Cleanup()
+        {
+            Directory.Delete(_workingDirectory, true);
+        }
+
+        [TestMethod]
+        public async Task LocFilesAreExported()
+        {
+            string[] exportedFiles = await CreateAndExportTemplateJson(
```

Comment:
imo, E2E tests should run the process instead. Can we utilize existing framework from dotnet3 integration tests to do it?

Reply:
Is that going to be better?
It is hard to debug a failing test when you are running a new process since you cannot step into it. Test runs also take longer because of the cost of starting a new process. Why is calling main bad?

Reply:
Actually I didn't realize that this is private method which calls Main instead - can we maybe rename the method to `ExecuteLocalizer...` or `RunLocalizer...` 
Also I'm not sure if there is a benefit in this method: it is short and called only twice; imo it would be better to inline it, then reading the test is more straight forward.

I would prefer to use existing framework as it is very "readable" and has a lot of good features in, redirect and asserting outputs etc.